src/types/ssh.ts

```diff
@@ -38,12 +40,28 @@ export type SshProvider<
   PR extends PluginSshRequest = PluginSshRequest,
   O extends object | undefined = undefined,
   SR extends SshRequest = SshRequest,
+  C extends object | undefined = undefined, // credentials object
 > = {
   requestToSsh: (request: CliPermissionSpec<PR, O>) => SR;
   toCliRequest: (
     request: Request<PR>,
     options?: { debug?: boolean }
   ) => Promise<Request<CliSshRequest>>;
+  cloudProviderLogin: (authn: Authn, request: SR) => Promise<C>;
+  proxyCommand: (request: SR) => string[];
+  reproCommands: (request: SR) => string[] | undefined;
```

Comment:
Can you please add jsdoc to each of these? I'm confused as to what `reproCommands` etc. are meant to be.